services/wfm/src/main/java/org/openkilda/wfm/topology/flow/service/FlowService.java

```diff
@@ -0,0 +1,213 @@
+        flowSegmentRepository = persistenceManager.getRepositoryFactory().createFlowSegmentRepository();
+        switchRepository = persistenceManager.getRepositoryFactory().createSwitchRepository();
+        islRepository = persistenceManager.getRepositoryFactory().createIslRepository();
+    }
+
+    public Iterable<Flow> getFlow(String flowId) {
+        return flowRepository.findById(flowId);
+    }
+
+    /**
+     * Stores flow and it's segments into DB.
+     *
+     * @param flowPair - forward and reverse flows to be saved.
+     */
+    public void createFlow(FlowPair flowPair) {
+        TransactionManager transactionManager = persistenceManager.getTransactionManager();
+        transactionManager.begin();
+        try {
+            createFlowForPair(flowPair);
+            transactionManager.commit();
+        } catch (Exception e) {
+            transactionManager.rollback();
+            throw e;
+        }
+    }
+
+    private void createFlowForPair(FlowPair flowPair) {
+        Flow flow = flowPair.getForward();
+        flowSegmentRepository.deleteFlowSegments(flow);
+        processCreateFlow(flow);
+        flow = flowPair.getReverse();
+        processCreateFlow(flow);
+    }
+
+    private void processCreateFlow(Flow flow) {
+        Switch srcSwitch = switchRepository.findBySwitchId(flow.getSrcSwitchId());
+        Switch dstSwitch = switchRepository.findBySwitchId(flow.getDestSwitchId());
+        flow.setSrcSwitch(srcSwitch);
+        flow.setDestSwitch(dstSwitch);
+        flow.setLastUpdated(Instant.now());
+        flowRepository.createOrUpdate(flow);
+        flowSegmentRepository.mergeFlowSegments(flow);
+        islRepository.updateIslBandwidth(flow);
+    }
+
+    /**
+     * Deletes flow and it's segments from DB.
+     *
+     * @param flowId - flow id to be removed.
+     */
+    public void deleteFlow(String flowId) {
+        TransactionManager transactionManager = persistenceManager.getTransactionManager();
+        transactionManager.begin();
+        try {
+            Iterable<Flow> flows = flowRepository.findById(flowId);
+            for (Flow flow : flows) {
+                processDeleteFlow(flow);
+            }
+            transactionManager.commit();
+        } catch (Exception e) {
+            transactionManager.rollback();
+            throw e;
+        }
+    }
+
+    /**
+     * Get FlowPair by the FlowId.
+     * @param flowId - flow identificator
+     * @return FlowPair object
+     */
+    public FlowPair getFlowPair(String flowId) {
+        Iterable<Flow> flows = flowRepository.findById(flowId);
+        FlowPair.FlowPairBuilder flowPairBuilder = FlowPair.builder();
+        for (Flow flow: flows) {
+            if (flow.isForward()) {
+                flowPairBuilder.forward(flow);
+            } else {
+                flowPairBuilder.reverse(flow);
+            }
+        }
+        return flowPairBuilder.build();
```

Comment:
Need validation that both "directions" are present.

Reply:
Done via notnull checks in Flow entity of the data model.